buildSrc/src/main/java/org/elasticsearch/gradle/ReaperPlugin.java

```diff
@@ -27,14 +28,21 @@ public void apply(Project project) {
 
         project.getPlugins().apply(GlobalBuildInfoPlugin.class);
 
-        Path inputDir = project.getRootDir()
+        File inputDir = project.getRootDir()
             .toPath()
             .resolve(".gradle")
             .resolve("reaper")
-            .resolve("build-" + ProcessHandle.current().pid());
-        ReaperService service = project.getExtensions()
-            .create("reaper", ReaperService.class, project, project.getBuildDir().toPath(), inputDir);
+            .resolve("build-" + ProcessHandle.current().pid())
+            .toFile();
 
-        project.getGradle().buildFinished(result -> service.shutdown());
+        var reaperServiceProvider = project.getGradle().getSharedServices().registerIfAbsent("reaper", ReaperService.class, spec -> {
+            // Provide some parameters
+            spec.getParameters().getInputDir().set(inputDir);
+            spec.getParameters().getBuildDir().set(project.getBuildDir());
+            spec.getParameters().getInternal().set(BuildParams.isInternal());
```

Comment:
I'm not sure there's any benefit to using a `Property` here. I think with managed types we can still use raw `boolean` which would just simplify stuff in the reaper service itself.

Reply:
done